### PR TITLE
refactor xiao a1 into modifier

### LINKED ISSUE #1000: xiao a1 refactored as a mod
image

right now the user must enable the `character` debug setting to see the effect of xiao's a1, because it is written directly into his snapshot AttackInfo [here](https://github.com/genshinsim/gcsim/blob/084b12073c0ea675b630177380e3363fb263f445/internal/characters/xiao/xiao.go#L84). it won't show in a list of damage mods which is perhaps confusing

### PR DESCRIPTION
**Pull request by skippi**
This fixes #1000.

### CODE REVIEW

File: internal/characters/xiao/burst.go [hunk 1]
```diff
@@ -30,13 +30,14 @@ func (c *char) Burst(p map[string]int) action.ActionInfo {
 	// TODO: No indication of that in library though
 	c.AddStatus(burstBuffKey, 900+burstStart, true)
```
Comment (skippi) [inline internal/characters/xiao/burst.go:31]: I find it a bit strange that the burst buff starts as soon as he casts. So in this case, does the A1 buff start ticking on cast too?

Comment (k0l11) [reply internal/characters/xiao/burst.go:31]: yes

File: internal/characters/xiao/asc.go [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+	"github.com/genshinsim/gcsim/pkg/core/combat"
+	"github.com/genshinsim/gcsim/pkg/core/player/character"
+	"github.com/genshinsim/gcsim/pkg/modifier"
+)
+
+const a1Key = "xiao-a1"
+
+func (c *char) a1() {
+	m := make([]float64, attributes.EndStatType)
+	c.AddAttackMod(character.AttackMod{
```
Comment (k0l11) [inline internal/characters/xiao/asc.go:14]: you aren't doing checks using atk and t, so no need to make this an AttackMod. should be a statmod instead

File: internal/characters/xiao/asc.go [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+import (
+	"github.com/genshinsim/gcsim/pkg/core/attributes"
+	"github.com/genshinsim/gcsim/pkg/core/combat"
+	"github.com/genshinsim/gcsim/pkg/core/player/character"
+	"github.com/genshinsim/gcsim/pkg/modifier"
+)
+
+const a1Key = "xiao-a1"
+
```
Comment (k0l11) [inline internal/characters/xiao/asc.go:11]: missing talent description comment

**Review by k0l11 — APPROVED** [review]

**Pull request merged**